pkg/controller/job/job_controller.go

```diff
@@ -1937,6 +1916,17 @@ func errorFromChannel(errCh <-chan error) error {
 	return nil
 }
 
+func parallelizeUntil(ctx context.Context, pieces int, doWorkPiece func(i int) error) error {
+	errCh := make(chan error, pieces)
+	workqueue.ParallelizeUntil(ctx, pieces, pieces, func(i int) {
+		if err := doWorkPiece(i); err != nil {
+			errCh <- err
+			utilruntime.HandleError(err)
+		}
+	})
+	return errorFromChannel(errCh)
```

Comment:
It might be better to do do an `Aggregate` error instead of the `HandleError` above and let the main loop handle the error

Reply:
Yes, I think so too. 
OTOH, the current implementation returns only one error that is arrived at errorCh, first.

So, there is no reasons for the current implementation. We want to aggregate all of errors into one, the return to the parent.